### PR TITLE
Add Administration Search

### PR DESCRIPTION
**Pull request by lucasxsong**
image

All administrations are being queried and stored client side (~150) in this update. Navigation between pages is noticeably snappier, and search results also return quickly. I observed a small increase in load time (2.5s to load on roar.education to 3.5s locally)

### CODE REVIEW

File: src/helpers/query/administrations.js
Comment (richford) [inline src/helpers/query/administrations.js:316]: Could we achieve the same thing by setting `page = 0` and `pageLimit = Number.MAX_SAFE_INTEGER` in the old `administrationPageFetcher`.

File: src/pages/HomeAdministrator.vue
Comment (richford) [inline src/pages/HomeAdministrator.vue:19]: Could we use Primevue's autocomplete component (https://primevue.org/autocomplete/) to suggest things as the user types?

**Review by richford — CHANGES_REQUESTED** [review]
@lucasxsong I love it! I made two comments that imply some requested changes:
- Can we avoid writing a new `administrationFetcher` function by passing in `page = 0` and `pageLimit = Number.MAX_SAFE_INTEGER` to the existing `administrationPageFetcher`?
- Can we use [Primevue's autocomplete component](https://primevue.org/autocomplete/) to suggest things as the user types?

**Comment by richford** [comment]
Additionally, could you add a Cypress test for searching by name?

**Review by richford — APPROVED** [review]
LGTM. Approved!. Let's wait for Cypress tests to pass and get a second approval from either @Emily-ejag or @ksmontville.

**Review by Emily-ejag — APPROVED** [review]

**Review by Emily-ejag — CHANGES_REQUESTED** [review]
It is not working for PartnerAdmins therefore the tests are failing

**Review by richford — CHANGES_REQUESTED** [review]
I was thinking about this a bit more and I'd like to request an additional refactor. Currently, with `onSortChange` we initiate a new query. However, I think we should remove the `orderBy` from the query key so that we don't invalidate the query by sorting. After all, we already have all of the administration data locally. So I think we should just sort locally without running another query.

- Can you change the `onSortChange` function to edit `filteredAdministrations` in place?
- Sorting and searching can also interact. When a user sorts and then searches, we should return the results in their previously requested sort order. Similarly, when a user searches, then sorts, we should only sort the administrations returned by the search pattern.
- Lastly, once @Emily-ejag merges #498, then partner admins will see the public facing name stored in the `publicName` field. Given that the Cypress tests are failing here, I think it might be best to merge her PR first. Can you work with @Emily-ejag to edit `onSortChange` to sort by `publicName` if the user is a partner admin and by `name` if the user is a super admin?
- Similarly, can you search by `publicName` if the user is a partner admin and `name` if the user is a super admin?
- In the two cases above, if an administration doesn't have the `publicName` field, it should fall back on the `name` field.

File: cypress/e2e/super-admin/searchAdministrations.spec.cy.js
Comment (ksmontville) [inline cypress/e2e/super-admin/searchAdministrations.spec.cy.js:2]:
```suggestion
const timeout = Cypress.env('timeout')
```

File: cypress/e2e/super-admin/searchAdministrations.spec.cy.js
Comment (ksmontville) [inline cypress/e2e/super-admin/searchAdministrations.spec.cy.js:11]:
```suggestion

```

**Review by ksmontville — CHANGES_REQUESTED** [review]

**Review by richford — APPROVED** [review]

**Review by Emily-ejag — APPROVED** [review]

**Review by ksmontville — APPROVED** [review]

**Pull request merged**
